src/trading/NewOrders.sol

```diff
@@ -0,0 +1,338 @@
+        if (_type == Trading.TradeTypes.Bid) { 
+            return (_fxpPrice < orders[_orderId].fxpPrice);
+        } else {
+            return (_fxpPrice > orders[_orderId].fxpPrice);
+        }
+    }
+
+    function assertIsNotBetterPrice(Trading.TradeTypes _type, IMarket _market, uint8 _outcome, uint256 _fxpPrice, bytes20 _betterOrderId) public constant returns (bool) { 
+        require(!isBetterPrice(_type, _market, _outcome, _fxpPrice, _betterOrderId));
+        return true;
+    }
+
+    function assertIsNotWorsePrice(Trading.TradeTypes _type, IMarket _market, uint8 _outcome, uint256 _fxpPrice, bytes20 _worseOrderId) public returns (bool) { 
+        require(!isWorsePrice(_type, _market, _outcome, _fxpPrice, _worseOrderId));
+        return true;
+    }
+
+    function insertOrderIntoList(bytes20 _orderId, Trading.TradeTypes _type, IMarket _market, uint8 _outcome, uint256 _fxpPrice, bytes20 _betterOrderId, bytes20 _worseOrderId) internal onlyWhitelistedCallers returns (bool) { 
+        bytes20 _bestOrderId = bestOrder[getBestOrderWorstOrderHash(_market, _outcome, _type)];
+        bytes20 _worstOrderId = worstOrder[getBestOrderWorstOrderHash(_market, _outcome, _type)];
+        NewOrdersFetcher _ordersFetcher = NewOrdersFetcher(controller.lookup("NewOrdersFetcher"));
+        (_betterOrderId, _worseOrderId) = _ordersFetcher.findBoundingOrders(_type, _market, _outcome, _fxpPrice, _bestOrderId, _worstOrderId, _betterOrderId, _worseOrderId);
+        if (_type == Trading.TradeTypes.Bid) {
+            _bestOrderId = updateBestBidOrder(_orderId, _market, _fxpPrice, _outcome);
+            _worstOrderId = updateWorstBidOrder(_orderId, _market, _fxpPrice, _outcome);
+        } else {
+            _bestOrderId = updateBestAskOrder(_orderId, _market, _fxpPrice, _outcome);
+            _worstOrderId = updateWorstAskOrder(_orderId, _market, _fxpPrice, _outcome);
+        }
+        if (_bestOrderId == _orderId) { 
+            _betterOrderId = 0;
+        }
+        if (_worstOrderId == _orderId) { 
+            _worseOrderId = 0;
+        }
+        if (_betterOrderId != 0) {
+            orders[_betterOrderId].worseOrderId = _orderId;
+            orders[_orderId].betterOrderId = _betterOrderId;
+        }
+        if (_worseOrderId != 0) {
+            orders[_worseOrderId].betterOrderId = _orderId;
+            orders[_orderId].worseOrderId = _worseOrderId;
+        }
+        return true;
+    }
+
+    // FIXME: Remove unused parameters orderID and gasPrice
+    function saveOrder(address orderId, Trading.TradeTypes _type, IMarket _market, uint256 _fxpAmount, uint256 _fxpPrice, address _sender, uint8 _outcome, uint256 _fxpMoneyEscrowed, uint256 _fxpSharesEscrowed, bytes20 _betterOrderId, bytes20 _worseOrderId, uint256 _tradeGroupId, uint256 gasPrice) public onlyWhitelistedCallers returns (bytes20 _orderId) { 
```

Comment:
For unused parameters I recommend removing the parameter name in order to silence the checker.  `saveOrder(address, Trading.TradeTypes _type, ..., uint256)` (note the missing parameter names for `orderId` and `gasPrice`.